### PR TITLE
Create test cases and win32 emu layer for bind()

### PR DESCRIPTION
**Pull request by maxz-lab**
Create tests and win32 emu layer for socket functions

The socket functions on win32 takes kernel handler instead
file descriptor, which requires a wrapper to match posix interface.

This change provides emulation for
    bind(), listen(), accept(), connect()

This change needs CL275526 for PS5 support.

b/302701164

Change-Id: I5b248c12ce28b6e9c8cc889c2bb3417107555be8

### CODE REVIEW

File: starboard/shared/win32/posix_emu/ifaddrs.cc [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  struct ifaddrs* ifaddr_ptr = nullptr;
+  struct ifaddrs* last_ifaddr_ptr = nullptr;
+
+  if (gethostname(ac, sizeof(ac)) == SOCKET_ERROR) {
+    return -1;
+  }
+
+  struct hostent* phe = gethostbyname(ac);
+  if (phe == 0) {
+    return -2;
```
Comment (y4vor) [inline starboard/shared/win32/posix_emu/ifaddrs.cc:46]: Please use -1

File: starboard/shared/win32/posix_emu/ifaddrs.cc [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  for (int i = 0; phe->h_addr_list[i] != 0; ++i) {
+    ifaddr_ptr = new struct ifaddrs;
```
Comment (y4vor) [inline starboard/shared/win32/posix_emu/ifaddrs.cc:50]: Please memset to zero the struct

File: starboard/shared/win32/posix_emu/include/sys/socket.h [hunk 3]
```diff
@@ -25,6 +26,25 @@ extern "C" {
 int sb_socket(int domain, int type, int protocol);
 #define socket sb_socket
 
+int sb_bind(int socket, const struct sockaddr* address, socklen_t address_len);
```
Comment (y4vor) [inline starboard/shared/win32/posix_emu/include/sys/socket.h:29]: Are `sockaddr` and `socket_t` ABI stable? Do we know the size and memory layout?

Comment (maxz-lab) [reply starboard/shared/win32/posix_emu/include/sys/socket.h:29]: sockaddr and sockaddr_in are idential in memory layout.
And socklen_t is an integer type of width of at least 32 bits.

File: starboard/nplb/posix_compliance/posix_socket_accept_test.cc [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+namespace {
+
+TEST(PosixSocketAcceptTest, RainyDayInvalidSocket) {
+  int invalid_socket_fd = -1;
+  EXPECT_FALSE(accept(invalid_socket_fd, NULL, NULL) == 0);
+}
+
+TEST(PosixSocketAcceptTest, RainyDayNoConnection) {
+  // Set up a socket to listen.
+  int socket_listen_fd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
```
[the remark this answers is not in the document]
Comment (maxz-lab) [reply starboard/nplb/posix_compliance/posix_socket_accept_test.cc:40]: Yes, they are portable.

**Review by y4vor — APPROVED** [review]

**Pull request merged**